ui/app/AppLayouts/Communities/popups/BurnTokensPopup.qml

```diff
@@ -99,53 +107,41 @@ StatusDialog {
                 StatusRadioButton {
                     id: specificAmountButton
 
+                    Layout.fillWidth: true
+
                     text: qsTr("Specific amount")
                     font.pixelSize: Style.current.primaryTextFontSize
                     ButtonGroup.group: radioGroup
 
                     onToggled: if(checked) amountToBurnInput.forceActiveFocus()
                 }
 
-                StatusInput {
-                    id: amountToBurnInput
+                AmountInput {
+                    id: amountInput
 
                     Layout.preferredWidth: 192
                     Layout.leftMargin: 30
-                    enabled: specificAmountButton.checked
-                    validationMode: StatusInput.ValidationMode.OnlyWhenDirty
-                    validators: [
-                        StatusValidator {
-                            validate: (value) => {
-                                const intAmount = parseInt(value)
-
-                                if (!intAmount)
-                                    return false
-
-                                const current = SQUtils.AmountsArithmetic.fromNumber(
-                                              intAmount, root.multiplierIndex)
-                                const remaining = SQUtils.AmountsArithmetic.fromString(
-                                              root.remainingTokens)
-
-                                return SQUtils.AmountsArithmetic.cmp(current, remaining) <= 0
-                            }
-                            errorMessage: qsTr("Exceeds available remaining")
-                        },
-                        StatusValidator {
-                            validate: (value) => { return parseInt(value) !== 0 }
-                            errorMessage: qsTr("Amount must be greater than 0")
-                        },
-                        StatusRegularExpressionValidator {
-                            regularExpression: Constants.regularExpressions.numerical
-                            errorMessage: qsTr("Invalid characters (0-9 only)")
-                        }
-                    ]
+                    customHeight: 44
+
+                    allowDecimals: root.multiplierIndex > 0
+                    maximumAmount: root.remainingTokens
+                    multiplierIndex: root.multiplierIndex
+
+                    validateMaximumAmount: true
+                    allowZero: false
+
+                    placeholderText: qsTr("Enter amount")
+                    labelText: ""
+
+                    maximumExceededErrorText: qsTr("Exceeds available remaining")
                 }
             }
 
             StatusRadioButton {
                 id: allTokensButton
 
                 Layout.alignment: Qt.AlignTop
+                Layout.fillWidth: true
```

Comment:
@micieslak WDYT of setting an explicit width for the popup? This way the binding loop will be fixed and we'll make sure it won't crash.

This `StatusRadioButton` is misbehaving and triggers the polish loop and crash.

When the text inside this Control is long enough and the popup is resized, the text line count will increase/decrease, triggering an update in the `ColumnLayout` height and Popup height. Somehow the combination of Control height change + Layout and Popup height change triggers a polish loop. But not always. It's not 10/10 repro. Sometimes the width change works fine.

Either way, we won't see this crash in the app as long as the app minimum width is larger than 600px.

Reply:
Thanks @alexjba for your input. Fixed by reducing number of layouts and setting explicit width taken from the design as you suggested.